packages/block-editor/src/components/global-styles/use-global-styles-output.js

```diff
@@ -780,6 +783,7 @@ export const toStyles = (
 			// Process styles for block support features with custom feature level
 			// CSS selectors set.
 			if ( featureSelectors ) {
+				// Mutates the passed styles object.
 				const featureDeclarations = getFeatureDeclarations(
```

Comment:
If this is the function mutating the object (I see a delete there), why don't we update the function to avoid the mutation instead of cloning the whole tree.

Reply:
The deletion/mutation is used to format some styles specially. There's code similar to this:
```js
// special case
if ( duotoneSelector ) {
  ruleset += `${ duotoneSelector } { filter: ${ styles.filter.duotone }; }`;
  delete styles.filter;
}

// standard case (loop inside `getStylesDeclarations`)
if ( styles.filter.duotone ) {
  ruleset += `${ selector } { filter: { ${ styles.filter.duotone }; }`;
}
```

This is how `getFeatureDeclarations` extracts (and deletes) certain styles, and outputs them under a special selector, and how code for duotone extracts the `filter.duotone` styles. The rest is left for the standard `getStylesDeclarations` to handle. 

To avoid the `delete` mutation, we'd have to rewrite this logic. Like building an exclude list of already processed paths and passing it to `getStylesDeclarations`.

We can continue cloning the styles object, but I think that cloning the entire `tree` object is too much. We're mutating only the styles extracted by `getNodesWithStyles`. So, it could be that function that clones the styles when calling `nodes.push( { styles: clonedStyles } )`. The `pickStyleKeys` could do the cloning.